### PR TITLE
Add a developing with spectral-cube docs page

### PR DESCRIPTION
**Pull request by e-koch**
Adding a page that lists out suggestions for incorporating spectral-cube in new packages, or adding custom functions to apply to cubes.

Also adds a bit more detail into how we use `dask.Array` under the hood.

### CODE REVIEW

File: docs/developing_with_spectralcube.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
+.. _doc_developersnotes:
+
+Notes for development using spectral-cube
+=========================================
+.. currentmodule:: spectral_cube
```
Comment (keflavich) [inline docs/developing_with_spectralcube.rst:5]: whooaaaa didn't know you could do this.

File: docs/developing_with_spectralcube.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+spectral-cube is flexible and can used within other packages for
+development beyond the core package's capabilities. Two significant strengths
+are the use of memory-mapping and the integration with `dask <https://dask.org/>`_
+(:ref:`doc_dask`) to efficiently handle larger than memory data.
+
+This page provides suggestions for developing using spectral-cube in other
```
Comment (keflavich) [inline docs/developing_with_spectralcube.rst:12]:
```suggestion
This page provides suggestions for software development using spectral-cube in other
```

File: docs/developing_with_spectralcube.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+Parallelizing operations
+------------------------
+
+Several operations implemented in :class:`SpectralCube` can be parallelized
+using the `joblib <https://joblib.readthedocs.io/en/latest/>`_ package. Builtin methods
+in :class:`SpectralCube` with the `parallel` keyword will enable using joblib.
+
+New methods can take advantage of these features by using creating custom functions
```
Comment (keflavich) [inline docs/developing_with_spectralcube.rst:40]:
```suggestion
New methods can take advantage of these features by creating custom functions
```

File: docs/developing_with_spectralcube.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+to pass to :meth:`SpectralCube.apply_function_parallel_spatial` and
+:meth:`SpectralCube.apply_function_parallel_spectral`. These methods expect
+functions with a data and mask array input, with optional `**kwargs` that can be
+passed and expect an output array of the same shape as the input.
```
Comment (keflavich) [inline docs/developing_with_spectralcube.rst:44]:
```suggestion
:meth:`SpectralCube.apply_function_parallel_spectral`. These methods accept
functions that take a data and mask array input, with optional `**kwargs`, 
and that return an output array of the same shape as the input.
```

**Review by keflavich — APPROVED** [review]
a few copy-edits.  Could probably use another reviewer later, but very good

File: docs/developing_with_spectralcube.rst (hunk 1 above)
Comment (e-koch) [reply docs/developing_with_spectralcube.rst:5]: Me either! I copy-pasted from docs Chris wrote.

**Pull request merged**